### DISCUSSION TITLE
Check if player comes through the proxy or not

### DISCUSSION THREAD

**Discussion by Distractic** [Minestom/Minestom#1625] [Help] [answered]
I have a Velocity proxy and my Minestom server.
I would like to know how I can refuse the direct connection to the minestom server because the player doesn't use the proxy redirection.

Have you some clues for that ?

**Comment by ZakShearman** [comment, marked as answer]
Just enable bungeeguard or velocity authentication so they have to provide a secret only given when going through the proxy.

**Reply by ZakShearman** [reply]
Velocity (recommended) is VelocityProxy.enable(secret).

Bungee cord: BungeeCordProxy.setBungeeGuardTokens(Set<String>)
